ossindex/internal/cache/cache.go

```diff
@@ -0,0 +1,172 @@
+//
+// Copyright 2020-present Sonatype Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// Package cache has definitions and functions for processing the OSS Index Feed
+package cache
+
+import (
+	"errors"
+	"os"
+	"os/user"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/recoilme/pudge"
+	"github.com/sonatype-nexus-community/nancy/customerrors"
+	. "github.com/sonatype-nexus-community/nancy/logger"
+	"github.com/sonatype-nexus-community/nancy/types"
+)
+
+// Cache is a struct with methods meant to be used for getting values from a DB cache
+// DBName can be used to override the actual database name, primarily for testing
+// TTL can be used to set the amount of time a specific object can live in the cache
+type Cache struct {
+	DBName string
+	TTL    time.Time
+}
+
+const dbDirName = "nancy"
+
+// DBValue is a local struct used for adding a TTL to a Coordinates struct
+type DBValue struct {
+	Coordinates types.Coordinate
+	TTL         int64
+}
+
+func (c *Cache) getDatabaseDirectory() (dbDir string) {
+	usr, err := user.Current()
+	customerrors.Check(err, "Error getting user home")
+
+	return path.Join(usr.HomeDir, types.OssIndexDirName, dbDirName, c.DBName)
```

Comment:
naming feels a little off here.
I would think that `dbDirName` would be the last part of a path of `getDatabaseDirectory`.

Also does the directory make sense as part of the `Cache` struct?

Reply:
Adjusted a few things: 760310f